utils/cache/filecache/filecache_test.go

```diff
@@ -0,0 +1,754 @@
+		defer goleak.VerifyNone(t)
+
+		fs := filesystem.NewFs(filesystem.StandardFS)
+		tmpSrcDir, err := fs.TempDirInTempDir("test-cache-src")
+		require.NoError(t, err)
+		defer func() { _ = fs.Rm(tmpSrcDir) }()
+
+		_ = filesystemtest.CreateTestFileTree(t, fs, tmpSrcDir, time.Now(), time.Now())
+		evictSrcContent, err := fs.Ls(tmpSrcDir)
+		require.NoError(t, err)
+
+		evictSuffix := "-evict"
+		for i, path := range evictSrcContent {
+			newBaseName := path + evictSuffix
+			newPath := filesystem.FilePathJoin(fs, tmpSrcDir, newBaseName)
+			oldPath := filesystem.FilePathJoin(fs, tmpSrcDir, path)
+			err := fs.Move(oldPath, newPath)
+			require.NoError(t, err)
+			evictSrcContent[i] = newBaseName
+		}
+
+		_ = filesystemtest.CreateTestFileTree(t, fs, tmpSrcDir, time.Now(), time.Now())
+		storeSrcContent, err := fs.LsWithExclusionPatterns(tmpSrcDir, fmt.Sprintf(".*%s", evictSuffix))
+		require.NoError(t, err)
+
+		tmpCacheDir, err := fs.TempDirInTempDir("test-cache")
+		require.NoError(t, err)
+		defer func() { _ = fs.Rm(tmpCacheDir) }()
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		config := DefaultFileCacheConfig()
+		config.CachePath = tmpCacheDir
+		cache, err := NewFsFileCache(ctx, fs, fs, tmpSrcDir, config)
+		require.NoError(t, err)
+
+		defer func() {
+			err = cache.Close(ctx)
+			require.NoError(t, err)
+			cancel()
+		}()
+
+		for _, path := range evictSrcContent {
+			err := cache.Store(ctx, path)
+			require.NoError(t, err)
+		}
+
+		g, gCtx := errgroup.WithContext(ctx)
+		g.Go(func() error {
+			for _, path := range storeSrcContent {
+				if err := cache.Store(gCtx, path); err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+
+		g.Go(func() error {
+			for _, path := range evictSrcContent {
+				if err := cache.Evict(gCtx, path); err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+
+		err = g.Wait()
+		require.NoError(t, err)
+
+		var cacheTree []string
+		err = fs.ListDirTree(tmpCacheDir, &cacheTree)
+		require.NoError(t, err)
+
+		var srcTree []string
+		err = fs.ListDirTreeWithContextAndExclusionPatterns(ctx, tmpSrcDir, &srcTree, fmt.Sprintf(".*%s", evictSuffix))
+		require.NoError(t, err)
+
+		require.Equal(t, len(srcTree), len(cacheTree), "Cache Dir and Src Dir have differnet number of content")
+
+		srcRelTree, err := fs.ConvertToRelativePath(tmpSrcDir, srcTree...)
+		require.NoError(t, err)
+
+		cacheRelTree, err := fs.ConvertToRelativePath(tmpCacheDir, cacheTree...)
+		require.NoError(t, err)
+
+		slices.Sort(srcRelTree)
+		slices.Sort(cacheRelTree)
+		require.Equal(t, srcRelTree, cacheRelTree, "Cache Dir and Src Dir have differnet contents")
+
+	})
+
+	t.Run("Restore_Evict", func(t *testing.T) {
+		defer goleak.VerifyNone(t)
+
+		fs := filesystem.NewFs(filesystem.StandardFS)
+		tmpCacheDir, err := fs.TempDirInTempDir("test-cache")
+		require.NoError(t, err)
+		defer func() { _ = fs.Rm(tmpCacheDir) }()
+
+		tmpSrcDir, err := fs.TempDirInTempDir("test-cache-src")
```

Comment:
in tests it is better to use `t.TmpDir()` to create temporary directories.

this lets the test suite itself handle the cleanup and you won't be left with any files if your defer function fails.

I think you should make this change everywhere